fenix/app/src/main/java/org/mozilla/fenix/compose/ext/Modifier.kt

```diff
@@ -49,3 +56,30 @@ fun Modifier.dashedBorder(
         )
     },
 )
+
+/**
+ * Used when clickable needs to be debounced to prevent rapid successive clicks
+ * from calling the onClick function.
+ *
+ * @param debounceInterval The length of time to wait between click events
+ * @param onClick Callback for when item this modifier effects is clicked
+ */
+fun Modifier.debouncedClickable(
+    debounceInterval: Long,
+    onClick: () -> Unit,
+) = composed(
+    factory = {
+        var lastClickTime: Long by remember { mutableStateOf(0) }
+
+        this.then(
+            Modifier.clickable(
+                onClick = {
+                    if (SystemClock.elapsedRealtime() - lastClickTime > debounceInterval) {
+                        onClick()
+                    }
+                    lastClickTime = SystemClock.elapsedRealtime()
```

Comment:
should this line be inside the `if`?

Reply:
I think the lastClickTime should be updated every click right?

Reply:
it should be every click that's successfully debounced right? otherwise, if the click was spammed, the `if` would never be true?

Reply:
That could potentially end up debouncing clicks forever, which we may not want in every case even if we theoretically want for the wallpapers case. I don't think we need to accommodate both cases though, and can just err towards the general case.

Reply:
ah I see what you mean, the debounce should be based on last time the click actually calls `onClick()` without being debounced.
I'll make the change to move it inside the loop.